manta-trusted-setup/src/groth16/ceremony/config/ppot.rs

```diff
@@ -387,33 +387,43 @@ pub async fn client_contribute<C>(
 ) -> Result<(), CeremonyError<C>>
 where
     C: Ceremony,
-    C::Challenge: DeserializeOwned,
+    C::Challenge: Debug + DeserializeOwned,
     C::Identifier: Serialize,
     C::Nonce: Clone + Debug + DeserializeOwned + Serialize,
     C::Signature: Serialize,
 {
-    const LOCK_TIME: u64 = 5;
     let term = Term::stdout();
-    client::contribute(
+    let response = client::contribute(
         signing_key,
         identifier,
         "http://localhost:8080",
-        |state| match state {
-            Update::Timeout => {
+        |metadata, state| match state {
+            Continue::Timeout => {
                 let _ = term.clear_last_lines(1);
                 println!("You have timed out. Waiting in queue again ...");
             },
-            Update::Position(position) => {
+            Continue::Position(position) => {
                 let _ = term.clear_last_lines(1);
-                println!(
-                    "Waiting in queue... There are {} people ahead of you. Estimated Waiting Time: {} minutes.",
-                    style(position).bold().red(),
-                    style(LOCK_TIME * position).bold().blue(),
-                );
+                if position <= u32::MAX.into() {
+                    println!(
+                        "Waiting in queue... There are {} people ahead of you. Estimated Waiting Time: {}.",
+                        style(position).bold().red(),
+                        style(format!("{:?}", metadata.contribution_time_limit * (position as u32))).bold().blue(),
+                    );
+                } else {
+                    println!(
+                        "Waiting in queue... There are many people ahead of you. Estimated Waiting Time: forever.",
+                    );
+                }
             },
         },
     )
-    .await
+    .await?;
+    println!(
+        "Success! You are contributor number {} with this challenge hash: {:?}.",
```

Comment:
Need to add a tweet suggestion here. Perhaps
" I made contribution {round} to the #MantaNetworkTrustedSetup ! My contribution's hash is {XXX} "
where XXX = hash( the challenge hashes for each circuit ).